src/main/java/com/onaio/steps/handler/activities/HouseholdServerStatusUpdater.java

```diff
@@ -0,0 +1,81 @@
+package com.onaio.steps.handler.activities;
+
+import android.app.ListActivity;
+import android.app.ProgressDialog;
+import android.os.AsyncTask;
+
+import com.onaio.steps.R;
+import com.onaio.steps.activities.BaseListActivity;
+import com.onaio.steps.helper.DatabaseHelper;
+import com.onaio.steps.model.Household;
+import com.onaio.steps.model.ServerStatus;
+
+import java.util.List;
+
+public class HouseholdServerStatusUpdater {
+
+    private static final String TAG = HouseholdServerStatusUpdater.class.getSimpleName();
+
+    private final ListActivity activity;
+    private final ProgressDialog pd;
+
+    public HouseholdServerStatusUpdater(ListActivity activity) {
+        this.activity = activity;
+        pd = new ProgressDialog(activity);
+        pd.setMessage(activity.getString(R.string.please_wait));
+        pd.setIndeterminate(true);
+        pd.setCancelable(false);
+        pd.setCanceledOnTouchOutside(false);
+    }
+
+
+    public void markAllSent() {
+        LoadAllPendingHousehold asyncTask = new LoadAllPendingHousehold();
+        asyncTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
+    }
+
+    private class LoadAllPendingHousehold extends AsyncTask<Void, Void, Boolean> {
+
+        private final DatabaseHelper db;
+
+        public LoadAllPendingHousehold() {
+            db = new DatabaseHelper(activity);
+        }
+
+        @Override
+        protected void onPreExecute() {
+            super.onPreExecute();
+            if (!pd.isShowing()) {
+                pd.show();
+            }
+        }
+
+        @Override
+        protected Boolean doInBackground(Void... voids) {
+
+            boolean needToRefresh = false;
+
+            List<Household> households = Household.findAllExceptServerStatus(new DatabaseHelper(activity), ServerStatus.SENT);
+
+            // process households
+            for (int i = 0; i < households.size(); i++) {
```

Comment:
If this was supposed to confirm with ODK Collect the household form status, confirming that the form was actually submitted without errors and it was selected during submission, then I believe that this is wrong. This is similar to `"UPDATE households set server_status = "SENT" WHERE server_status != 'SENT'"`

Reply:
Make the changes to confirm from the ODK Collect content provider. Update the test also

Reply:
https://github.com/onaio/steps-app/pull/137#issuecomment-1022224662

Reply:
Makes sense